File: CSharp/101. Symmetric Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsSymmetric(TreeNode root) {
        return isSubTreeValid(root.left, root.right);
    }
    
    public bool isSubTreeValid(TreeNode a, TreeNode b){
        if(a == null && b == null){
            return true;
        }
        
        if((a == null && b != null) || (a != null && b == null)){
            return false;
        }
        if(a.val != b.val){
            return false;
        }
        
        return isSubTreeValid(a.left, b.right) && isSubTreeValid(a.right , b.left);
    }
}

File: CPP/101. Symmetric Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return isSubTreeValid(root -> left , root -> right);
    }
    
    bool isSubTreeValid(TreeNode *a, TreeNode *b){
        if(a == NULL && b == NULL){
            return true;
        }
        
        if((a == NULL && b != NULL) || (a != NULL && b == NULL)){
            return false;
        }
        if(a -> val != b -> val){
            return false;
        }
        
        return isSubTreeValid(a -> left, b -> right) && isSubTreeValid(a -> right , b -> left);
    }
};